db/db_basic_test.cc

```diff
@@ -2363,6 +2396,240 @@ INSTANTIATE_TEST_CASE_P(ParallelIO, DBBasicTestWithParallelIO,
                                            ::testing::Bool(), ::testing::Bool(),
                                            ::testing::Values(1, 4)));
 
+// A test class for intercepting random reads and injecting artificial
+// delays. Used for testing the deadline/timeout feature
+class DBBasicTestMultiGetDeadline : public DBBasicTestMultiGet {
+ public:
+  DBBasicTestMultiGetDeadline()
+      : DBBasicTestMultiGet("db_basic_test_multiget_deadline", 10, false, true,
+                            true, true, 1) {}
+
+  // Forward declaration
+  class DeadlineFS;
+
+  class DeadlineRandomAccessFile : public FSRandomAccessFileWrapper {
+   public:
+    DeadlineRandomAccessFile(DeadlineFS& fs,
+                             std::unique_ptr<FSRandomAccessFile>& file)
+        : FSRandomAccessFileWrapper(file.get()),
+          fs_(fs),
+          file_(std::move(file)) {}
+
+    IOStatus Read(uint64_t offset, size_t len, const IOOptions& opts,
+                  Slice* result, char* scratch, IODebugContext* dbg) const {
+      int delay;
+      if (fs_.ShouldDelay(&delay)) {
+        Env::Default()->SleepForMicroseconds(delay);
```

Comment:
I didn't understand the concern. What is the problem with a longer delay?

Reply:
There are limits on how long a unit test can run. If a unit test passes a large value to the deadline FS, and this will cause the test to sleep for a long time. I don't think it is necessary to sleep for long time, can we add a check to avoid this entirely?

Reply:
Got it. I'll assert if its greater than, say, 500ms. Just want to give enough buffer to ensure there's no flakiness.